.github/workflows/main.yml

```diff
@@ -68,8 +68,10 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
       - uses: eifinger/setup-rye@v2.6
+      - name: Pin python-version ${{ matrix.python-version }}
+        run: rye pin ${{ matrix.python-version }}
       - run: rye config --set-bool behavior.use-uv=true
-      - run: rye sync --no-lock
+      - run: rye sync
```

Comment:
This ignores the lockfile we have, because it generates a new one.

Reply:
Surely this uses the lockfile in Python 3.12? It doesn't have to be in this PR but I still think we should be testing without a lockfile as much as possible.